html/js/app.js

```diff
@@ -529,23 +530,31 @@ function FormatItemInfo(itemData, dom) {
                 itemData.info.ammo != null ? itemData.info.ammo : 0;
             }
             if (itemData.info.attachments != null) {
-                var attachmentString = "";
-                $.each(itemData.info.attachments, function(i, attachment) {
-                    if (i == itemData.info.attachments.length - 1) {
-                        attachmentString += attachment.label;
-                    } else {
-                        attachmentString += attachment.label + ", ";
-                    }
-                });
-                $(".item-info-description").html(
-                    "<p><strong>Serial: </strong><span>" +
-                    itemData.info.serie +
-                    "</span></p><p><strong>Ammo: </strong><span>" +
-                    itemData.info.ammo +
-                    "</span></p><p><strong>Attachments: </strong><span>" +
-                    attachmentString +
-                    "</span></p>"
-                );
+                $.post(
+                    "https://ps-inventory/GetWeaponData",
+                    JSON.stringify({
+                        weapon: itemData.name,
+                        ItemData: itemData,
+                    }),
+                    function(data) {
+                        var attachmentString = "";
+                        $.each(data.AttachmentData, function(i, attachment) {
+                            if (i == itemData.info.attachments.length - 1) {
+                                attachmentString += attachment.label;
+                            } else {
+                                attachmentString += attachment.label + ", ";
+                            }
+                        });
+                        $(".item-info-description").html(
+                            "<p><strong>Serial: </strong><span>" +
+                            itemData.info.serie +
+                            "</span></p><p><strong>Ammo: </strong><span>" +
+                            itemData.info.ammo +
+                            "</span></p><p><strong>Attachments: </strong><span>" +
+                            attachmentString +
+                            "</span></p>"
+                        );
```

Comment:
Inserting html like this directly on an HTML element can be quite dangerous for exploding the code. A better solution would be to build up a couple of elements and then appending them to the `item-info-description` element. It could look somewhat like this:

```js
const description = document.createElement("p");
const header = document.createElement("strong");
const value = document.createElement("span");

header.innerText = "Serial: ";
value.innerText = item.info.ammo;
description.innerHTML = `${header} {value}`

document.querySelector(".info-info-description").append(description)
```
You can then loop over all the values inside the `itemData.info` array to output them all. Requires a little fiddling with the code above.

Reply:
this was default code from the old ps-inv just made it compat with new qb-core